### PR TITLE
Feature/update phoenix 1.2.x

### PR DESCRIPTION
**Pull request by silbermm**
This is to upgrade Phoenix to 1.2 per [these instructions](http://www.phoenixframework.org/blog/upgrading-from-11x-to-120)

The biggest change was in the handling of forwarded routes. I had to create scopes for the microclients and use `get /*path ...` instead of `forward /connect ...` . I believe this warrants some regression testing.

### CODE REVIEW

File: apps/crossroads_interface/web/views/error_helpers.ex [hunk 1]
```diff
@@ -14,22 +14,24 @@ defmodule CrossroadsInterface.ErrorHelpers do
...
     #
     #     dngettext "errors", "1 file", "%{count} files", count
+    #     dgettext "errors", "is invalid"
     #
-    Gettext.dngettext(CrossroadsInterface.Gettext, "errors", msg, msg, opts[:count], opts)
-  end
-
-  def translate_error(msg) do
-    Gettext.dgettext(CrossroadsInterface.Gettext, "errors", msg)
+    if count = opts[:count] do
```
Comment (mjvcallibrity) [inline apps/crossroads_interface/web/views/error_helpers.ex]: Is this supposed to be variable assignment or comparison?

File: apps/crossroads_interface/mix.exs [hunk 2]
```diff
@@ -22,15 +22,16 @@ defmodule CrossroadsInterface.Mixfile do
 
   def application do
     [mod: {CrossroadsInterface, []},
-     applications: [:phoenix, :phoenix_html, :logger_file_backend, :cowboy, :logger, :gettext, :crossroads_content, :ssl]]
+     applications: [:phoenix, :phoenix_pubsub, :phoenix_html, :logger_file_backend, :cowboy, :logger, :gettext, :crossroads_content, :ssl]]
   end
```
Comment (richardbarker) [inline apps/crossroads_interface/mix.exs]: Do we need to add phoenix_pubsub in order to upgrade?  We aren't using it as far as I know.

File: apps/crossroads_content/lib/content_pages.ex [hunk 3]
```diff
@@ -82,7 +80,7 @@ defmodule CrossroadsContent.Pages do
     {:ok, %{}}
   end
 
-  def handle_info(:refresh_cms_page_cache, cms_page_cache) do
+  def handle_info(:refresh_cms_page_cache, _cms_page_cache) do
```
Comment (timmyg) [inline apps/crossroads_content/lib/content_pages.ex]: is _cms_page_cache var used?

File: apps/crossroads_interface/web/controllers/proxy_content_controller.ex [hunk 4]
```diff
@@ -1,68 +0,0 @@
-defmodule CrossroadsInterface.ProxyContentController do
```
Comment (timmyg) [inline apps/crossroads_interface/web/controllers/proxy_content_controller.ex]: were these files moved or not needed?

File: apps/crossroads_content/lib/content_pages.ex (hunk 3 above)
Comment (timmyg) [reply apps/crossroads_content/lib/content_pages.ex]: actually after looking at your other changes, assuming this is just the local variable convention that should be used

File: apps/crossroads_interface/web/controllers/proxy_content_controller.ex (hunk 4 above)
Comment (silbermm) [reply apps/crossroads_interface/web/controllers/proxy_content_controller.ex]: Not needed. They were added before we deployed to Apache . Now Apache is handling the proxy functions.

File: apps/crossroads_interface/mix.exs [hunk 5]
```diff
@@ -23,24 +23,25 @@ defmodule CrossroadsInterface.Mixfile do
...
 
   defp elixirc_paths(:test), do: ["lib", "web", "test/support"]
   defp elixirc_paths(_),     do: ["lib", "web"]
 
   defp deps do
-    [{:phoenix, "~> 1.1.3"},
-     {:phoenix_html, "~> 2.3"},
+    [{:phoenix, "~> 1.2.0"},
+     {:phoenix_html, "~> 2.6"},
+     {:phoenix_pubsub, "~> 1.0"},
```
Comment (richardbarker) [inline apps/crossroads_interface/mix.exs]: Do we need to include phoenix_pubsub to upgrade the version?  We aren't using it are we?

Comment (silbermm) [reply apps/crossroads_interface/mix.exs]: probably not... I'll take it out and make sure nothing breaks.

**Pull request merged**